doc/diagnostics/WritingDdrCode.md

```diff
@@ -0,0 +1,174 @@
+```
+
+Pointer classes generated in source form during the build are unaffected by this extra field
+information. Those generated in binary form (i.e. in `.class` files) are limited so they only
+include methods corresponding to fields mentioned in `AuxFieldInfo29.dat` to prevent inadvertent
+use of a field (that has not always been present) in hand-written code.
+
+The source pattern for optional fields was modified to throw a checked exception (`NoSuchFieldException`):
+
+```
+    // U8* realHeapTop
+    @com.ibm.j9ddr.GeneratedFieldAccessor(offsetFieldName="_realHeapTopOffset_", declaredType="U8*")
+    public U8Pointer realHeapTop() throws CorruptDataException, NoSuchFieldException {
+        try {
+            return U8Pointer.cast(getPointerAtOffset(J9ModronThreadLocalHeap._realHeapTopOffset_));
+        } catch (NoClassDefFoundError | NoSuchFieldError e) {
+            throw new NoSuchFieldException();
+        }
+    }
+
+    // U8* realHeapTop
+    public PointerPointer realHeapTopEA() throws CorruptDataException, NoSuchFieldException {
+        try {
+            return PointerPointer.cast(nonNullFieldEA(J9ModronThreadLocalHeap._realHeapTopOffset_));
+        } catch (NoClassDefFoundError | NoSuchFieldError e) {
+            throw new NoSuchFieldException();
+        }
+    }
+```
+
+Dynamically generated classes include either this when the field is present (in a 64-bit core):
+
+```
+    public U8Pointer realHeapTop() throws CorruptDataException, NoSuchFieldException {
+        return U8Pointer.cast(getPointerAtOffset(8));
+    }
+```
+
+or this when the field is absent:
+
+```
+    public U8Pointer realHeapTop() throws CorruptDataException, NoSuchFieldException {
+        throw new NoSuchFieldException();
+    }
+```
+
+Code that accesses optional fields must handle `NoSuchFieldException` because it is a checked
+exception.
+
+To start using a field in DDR code, an entry must be added to `AuxFieldInfo29.dat`. If the field
+has always been present, then it may be marked as `required`; otherwise, specify its type and
+handle the possibility of `NoSuchFieldException` where it is used.
+
+If a field required by DDR code is removed, the word `required` (in `AuxFieldInfo29.dat`)
+must be replaced by the type of the field (as previously recorded in `j9ddr.dat`) and DDR code
+must be updated to deal appropriately with `NoSuchFieldException`.
+
+A similar problem arises with the use of constants. The value of a constant may change over
+time without issue: its value will be captured in the DDR blob found within core files (a
+copy of the contents of `j9ddr.dat`). However, constants that have not always existed can
+translate to `NoSuchFieldError` when accessed in DDR code.
+
+The solution, introduced by [#12210](https://github.com/eclipse-openj9/openj9/pull/12210),
+involves two new files: `superset-constants.dat` and `CompatibilityConstants29.dat`.
+
+The first file, `superset-constants.dat`, lists constants that are legal to be used in DDR
```

Comment:
A version of this paragraph is also needed for the TLDR section.

Reply:
Done.